Tests/LibWeb/Ref/expected/render-order-floating-replaced-ref.html

```diff
@@ -0,0 +1,27 @@
+<!DOCTYPE html>
```

Comment:
I compared this page on build from master vs other browsers and it seems to visually already be painted correctly. could we add a test that demoes correctness improvement?

Reply:
I guess something changed then in the commits on master that I rebased over. It rendered incorrectly on master when I first opened the PR. I'll take a look.

Reply:
The test still reproduces the original problem for me as of 29d74632c773048987e0a5a708911eed68645770.

| Master | This PR/other browser |
|--------|--------|
| ![wrong](https://github.com/user-attachments/assets/508b183d-4d89-4622-bf8a-0e2192cce6eb) | ![correct](https://github.com/user-attachments/assets/94672c0f-3852-4f4a-8a52-8e61df375b12) |

I've rebased on that commit now.

Reply:
@kalenikaliaksandr Does the test work correctly for you now (i.e. render as in the screenshots above)?